base/math.jl

```diff
@@ -804,47 +785,47 @@ function rem2pi end
 function rem2pi(x::Float64, ::RoundingMode{:Nearest})
     abs(x) < pi && return x
 
-    (n,y) = ieee754_rem_pio2(x)
+    n,y1,y2 = rem_pio2_kernel(x)
 
     if iseven(n)
         if n & 2 == 2 # n % 4 == 2: add/subtract pi
-            if y[1] <= 0
-                return add22condh(y[1],y[2],pi2o2_h,pi2o2_l)
+            if y1 <= 0
+                return add22condh(y1,y2,pi2o2_h,pi2o2_l)
             else
-                return add22condh(y[1],y[2],-pi2o2_h,-pi2o2_l)
+                return add22condh(y1,y2,-pi2o2_h,-pi2o2_l)
             end
         else          # n % 4 == 0: add 0
-            return y[1]
+            return y1+y2
```

Comment:
This was a bug. If you look at https://github.com/JuliaLang/julia/blob/c463c1f49872ac3d1b7c0809d00288c119ed0909/base/math.jl#L730 you can see that the offending lines are trying to be fast paths for `add22condh(y1, y2, 0.0, 0.0)`. This would then be
```
    r = xh+0.0 # xh
    s = (0.0 > 0.0) ? (xh-r+0.0+0.0+xl) : (0.0-r+xh+xl+0.0) # this is xl as r == xh
    zh = r+s # xh + xl
    return zh
```
so basically we can just return `y1+y2` directly.

Are you asking because you'd like a test added, a separate commit for this change, or something else?